lib/destination/types/table_config_test.go

```diff
@@ -130,26 +131,55 @@ func (t *TypesTestSuite) TestDwhTableConfig_ReadOnlyColumnsToDelete() {
 	wg.Wait()
 }
 
-func (t *TypesTestSuite) TestDwhTableConfig_ClearColumnsToDeleteByColName() {
-	colsToDelete := make(map[string]time.Time)
-	for _, colToDelete := range []string{"a", "b", "c", "d"} {
-		colsToDelete[colToDelete] = time.Now()
+func (t *TypesTestSuite) TestAuditColumnsToDelete() {
+	type _tc struct {
+		colsToDelete       []string
+		dropDeletedCols    bool
+		expectedColsRemain []string
 	}
 
-	tc := NewDwhTableConfig(nil, colsToDelete, false, false)
-	var wg sync.WaitGroup
-	assert.Equal(t.T(), 4, len(tc.columnsToDelete))
-	for _, colToDelete := range []string{"a", "b", "c", "d"} {
-		for i := 0; i < 100; i++ {
-			wg.Add(1)
-			go func(colName string) {
-				time.Sleep(time.Duration(jitter.JitterMs(50, 1)) * time.Millisecond)
-				defer wg.Done()
-				tc.ClearColumnsToDeleteByColName(colName)
-			}(colToDelete)
-		}
+	colsToDeleteList := []string{"aa", "ba", "ca", "da"}
+	tcs := []_tc{
+		{
+			colsToDelete:       colsToDeleteList,
```

Comment:
Might want to pass something else in for `colsToDelete`, otherwise the test would still pass even if `dropDeletedCols` was being ignored in `AuditColumnsToDelete`.